packages/fhir-converter/src/lib/outputProcessor/resourceDeduplicator.js

```diff
@@ -0,0 +1,98 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) 2022-present Metriport Inc.
+//
+// Licensed under AGPLv3. See LICENSE in the repo root for license information.
+//
+// This file incorporates work covered by the following copyright and
+// permission notice:
+//
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+//
+//     Permission to use, copy, modify, and/or distribute this software
+//     for any purpose with or without fee is hereby granted, provided
+//     that the above copyright notice and this permission notice appear
+//     in all copies.
+//
+//     THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+//     WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+//     WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+//     AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+//     CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
+//     OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+//     NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+//     CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+// -------------------------------------------------------------------------------------------------
+
+var merge = require("./resourceMerger").merge;
+
+module.exports.Process = function (jsonObj) {
+  try {
+    const encounters = jsonObj.entry.filter(entry => {
+      return entry.resource.resourceType === "Encounter";
+    });
+
+    const idMap = new Set();
+    let duplicateId;
+    encounters.forEach(enc => {
+      enc.resource.identifier.forEach(id => {
+        if (idMap.has(id.value)) {
+          duplicateId = id.value;
```

Comment:
Iterating over all ids and overwriting duplicateId each time and then Outside the loop only processing logic based on the single duplicate Id? Are you sure the logic is right here?

Reply:
You might want to push duplicateId into a list or add a duplicate field in the hash table. The latter is most efficient

Reply:
Thinking about it, I guess you're assuming here there is only one duplicated set of encounter IDs? Let's not assume that though since it might not be true plus makes the code have weird implicit assumptions

Reply:
Good point. But yea, it looks like there's ever only one pair of duplicated encounters that we're dealing with

Reply:
Ok but still change the logic right to reflect the case where there are multiple?

Reply:
Yea will do